addon/test-helper/deprecation.js

```diff
@@ -1,53 +1,50 @@
-/* globals QUnit */
-
-import MethodCallExpectation from "./method-call-expectation";
-
-var NONE = function(){};
+import MethodCallExpectation from './method-call-expectation';
+import MethodCallTracker from './method-call-tracker';
 
 var DeprecationAssert = function(env){
   this.env = env;
-
   this.reset();
 };
 
 DeprecationAssert.prototype = {
-
   reset: function(){
-    this.expecteds = null;
-    this.actuals = null;
+    this.tracker = new MethodCallTracker(this.env.Ember, 'deprecate');
```

Comment:
I have nothing specific in mind beyond trying to keep only one tracker around for the sake on sanity.

IMO could be as simple as having `invokeWithNewContext` call a `buildTracker` method that returns one present as `this.tracker` if already built and creates a new one (and sets it) if there is none present. Then `reset` could just tear it down.

In general, I encourage learning toward patterns that are lazy. create a call tracker only if needed, and use reset to tear it down and set the stage for the next if you need to.

Vague thoughts, I know.